dem_handler/utils/spatial.py

```diff
@@ -509,3 +514,61 @@ def crop_datasets_to_bounds(
             # )
 
     return dem_array, dem_profile
+
+
+def check_dem_type_in_bounds(dem_type: str, bounds: BBox) -> bool:
+    """Check if the specified dem has data within the provided bounds. The provided dem_type is matched to either the
+    Copernicus Global 30m DEM or REMA DEM (currently implemented options). True is returned if the provided bounds
+    intersect with any tiles of the specified DEM.
+
+    Parameters
+    ----------
+    dem_type : str
+        dem type to check. Can be variations of REMA and COP_30. e.g. REMA_32, REMA_10,
+        cop30m, cop_30m, cop_glo30.
+    bounds : BBox
+        Bounds to check if data exists
+
+    Returns
+    -------
+    bool
+        True if the bounds intersects a tile, False otherwise.
+
+    Raises
+    -------
+    ValueError
+        If the provided dem_type cannot be matched to either the Copernicus 30m global DEM or the REMA dem.
+    """
+
+    if type(bounds) == BoundingBox:
+        bounds = bounds.bounds
+
+    dem_type_match = dem_type.upper()
+    if "COP" and "30" in dem_type_match:
+        dem_index_path = COP30_GPKG_PATH
+        dem_type_formal = "Copernicus 30m global DEM"
+    elif "REMA" in dem_type_match:
```

Comment:
I'm wondering if our REMA logic might need to be more nuanced. The REMA geopackage has three different layers in it -- one each for 2m, 10m, 32m. It is possible to specify the layer on read using `df = gpd.read_file(dem_index_path, layer='REMA_Mosaic_Index_v2_32m')`. The copdem geopackage only has one layer, so there's no need to specify a layer.

The alternative is to export each REMA layer as its own geopackage, and then optionally choose to only support the 32m one at this time, or support all three.

Reply:
You might also need to be careful with logic, since if you did a case of `"REMA" and "2" in dem_type_match` that could be valid for either the 32m DEM or the 2m DEM.